src/Transpiler.ts

```diff
@@ -128,6 +137,18 @@ export abstract class LuaTranspiler {
         this.exportStack[this.exportStack.length - 1].push({name: nameIn, node: nodeIn, dummy: dummyIn});
     }
 
+    public peekSpecialScope(): SpecialScope {
+        return this.scopeStack[this.scopeStack.length - 1];
+    }
+
+    public pushSpecialScope(scopeType: ScopeType): void {
+        this.scopeStack.push({ type: scopeType, id: this.genVarCounter });
```

Comment:
You should probably increment genVarCounter after you use it, to prevent duplicates here